test/parallel/test-fs-open.js

```diff
@@ -2,18 +2,20 @@
 const common = require('../common');
 const assert = require('assert');
 const fs = require('fs');
+const path = require('path');
 
-let caughtException = false;
+const file = path.join(common.fixturesDir, 'file', 'does', 'not', 'exist');
+const winFile = file.replace(/\\/g, '\\\\');
+const expected = new RegExp('^Error: ENOENT: no such file or directory, ' +
+                            `open '${(common.isWindows) ? winFile : file}'$`);
 
-try {
-  // should throw ENOENT, not EBADF
-  // see https://github.com/joyent/node/pull/1228
-  fs.openSync('/path/to/file/that/does/not/exist', 'r');
-} catch (e) {
-  assert.strictEqual(e.code, 'ENOENT');
-  caughtException = true;
-}
-assert.strictEqual(caughtException, true);
+assert.throws(() => {
+  fs.openSync(file, 'r');
+}, (err) => {
+  assert.strictEqual(err.code, 'ENOENT');
+  assert(expected.test(err));
```

Comment:
Why using the regexp now that we are using a function? Can't we use

```js
assert.strictEqual(err.message, '...');
```

?

If we also want to check that `err` is an instance of `Error`, we can do that with one more assertion.